File: typescript/problems/1/1.ts
function twoSum(nums: number[], target: number): number[] {
  const map = new Map()
  for (const [idx, num] of nums.entries()) {
    if (map.has(target - num)) {
      return [idx, map.get(target - num)]
    }
    map.set(num, idx)
  }
}

File: leetcode-hot-100/python/1.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for index, num in enumerate(nums):
            anthor_num = target - num
            if anthor_num in hashmap:
                return [hashmap[anthor_num], index]
            hashmap[num] = index
        return None
